prow/crier/metrics.go

```diff
@@ -0,0 +1,39 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package crier reports finished prowjob status to git providers.
+package crier
+
+import "github.com/prometheus/client_golang/prometheus"
+
+// Prometheus Metrics
+var (
+	crierMetrics = struct {
+		latency *prometheus.HistogramVec
+	}{
+		latency: prometheus.NewHistogramVec(prometheus.HistogramOpts{
+			Name:    "report_time",
+			Help:    "Histogram of time spent reporting, calculated by the time difference between job completion and end of reporting.",
+			Buckets: []float64{1, 10, 60, 600, 1800, 3600},
```

Comment:
We probably want some more buckets between 1 and 600 seconds, right? I'd expect most to fall into either the 10 or 60 bucket during normal operation which won't tell us very much. I think we'll want to be able to measure improvements more granularly than that? 
Since we've limited the label set to just `reporter` its safe to use more buckets again.

Reply:
good point, added